semeio/workflows/misfit_preprocessor/config.py

```diff
@@ -136,106 +136,107 @@ def _inject_default_t(config):
     return config
 
 
-_LINKAGE_SCHEMA = {
+HIERARCHICAL_SCHEMA = {
     cs.MetaKeys.Type: cs.types.NamedDict,
-    cs.MetaKeys.Description: (
-        "The {linkage} implementation is backed by scipy and for a "
-        "more detailed description we refer the reader to the "
-        "documentation of scipy.cluster.hierarchy.linkage` "
-        "(https://docs.scipy.org/doc/scipy/reference/generated"
-        "/scipy.cluster.hierarchy.linkage.html)."
-    ).format(linkage=_LINKAGE),
     cs.MetaKeys.Content: {
-        _METHOD: {
-            cs.MetaKeys.Type: cs.types.String,
+        "hierarchical": {
+            cs.MetaKeys.ElementValidators: (_t_is_int_if_maxclust,),
+            cs.MetaKeys.Transformation: _inject_default_t,
+            cs.MetaKeys.Type: cs.types.NamedDict,
             cs.MetaKeys.Description: (
-                "Method used to calculate the distance between the clusters."
-            ),
-            cs.MetaKeys.ElementValidators: (_one_of(*_METHODS),),
-            cs.MetaKeys.Default: _AVERAGE,
-        },
-        _METRIC: {
-            cs.MetaKeys.Type: cs.types.String,
-            cs.MetaKeys.Description: ("Distance metric used to calculate distances."),
-            cs.MetaKeys.ElementValidators: (_one_of(*_METRICS),),
-            cs.MetaKeys.Default: _EUCLIDEAN,
-        },
+                "The {linkage} implementation is backed by scipy and for a "
+                "more detailed description we refer the reader to the "
+                "documentation of scipy.cluster.hierarchy.linkage` "
+                "(https://docs.scipy.org/doc/scipy/reference/generated"
+                "/scipy.cluster.hierarchy.linkage.html)."
+                "The {fcluster} implementation is backed by scipy and for a "
+                "more detailed description we refer the reader to the "
+                "documentation of scipy.cluster.hierarchy.fcluster` "
+                "(https://docs.scipy.org/doc/scipy/reference/generated"
+                "/scipy.cluster.hierarchy.fcluster.html)."
+            ).format(linkage=_LINKAGE, fcluster=_FCLUSTER),
+            cs.MetaKeys.Content: {
+                _SPEARMAN_THRESHOLD: {
+                    cs.MetaKeys.Type: cs.types.Number,
+                    cs.MetaKeys.Description: (
```

Comment:
Feel free to refactor to: f"String {variable}" syntax if you like (goes for all descriptions)

Reply:
I found inly this one that needed to be changed :tada:

Reply:
Would it be ok to change `raise ValidationError` format also?